src/scalar.rs

```diff
@@ -37,6 +37,12 @@ impl fmt::Display for Scalar {
     }
 }
 
+impl From<u32> for Scalar {
+    fn from(val: u32) -> Scalar {
+        Scalar([val as u64, 0, 0, 0]) * R2
+    }
+}
+
```

Comment:
As is visible from other changes in the PR, this decreases usability by requiring the caller to specify the source type (e.g. `5u64.into()` instead of `5.into()`). I suspect this would be more usable if `impl From<u64> for Scalar` was replaced with `impl From<T: Into<u64>> for Scalar` (and it would then also work for any `u*` type).